src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCacheResult.cs

```diff
@@ -17,29 +20,42 @@ public struct CompilerCacheResult
         /// Initializes a new instance of <see cref="CompilerCacheResult"/> with the specified
         /// <see cref="Compilation.CompilationResult"/>.
         /// </summary>
+        /// <param name="relativePath">Application relative path to the view file.</param>
         /// <param name="compilationResult">The <see cref="Compilation.CompilationResult"/>.</param>
-        public CompilerCacheResult(CompilationResult compilationResult)
-            : this(compilationResult, new IChangeToken[0])
+        public CompilerCacheResult(string relativePath, CompilationResult compilationResult)
+            : this(relativePath, compilationResult, new IChangeToken[0])
         {
         }
 
         /// <summary>
         /// Initializes a new instance of <see cref="CompilerCacheResult"/> with the specified
         /// <see cref="Compilation.CompilationResult"/>.
         /// </summary>
+        /// <param name="relativePath">Application relative path to the view file.</param>
         /// <param name="compilationResult">The <see cref="Compilation.CompilationResult"/>.</param>
         /// <param name="expirationTokens">One or more <see cref="IChangeToken"/> instances that indicate when
         /// this result has expired.</param>
-        public CompilerCacheResult(CompilationResult compilationResult, IList<IChangeToken> expirationTokens)
+        public CompilerCacheResult(string relativePath, CompilationResult compilationResult, IList<IChangeToken> expirationTokens)
         {
             if (expirationTokens == null)
             {
                 throw new ArgumentNullException(nameof(expirationTokens));
             }
 
-            CompilationResult = compilationResult;
-            Success = true;
             ExpirationTokens = expirationTokens;
+            var compiledType = compilationResult.CompiledType;
+
+            var newExpression = Expression.New(compiledType);
+
+            var pathProperty = compiledType.GetProperty(nameof(IRazorPage.Path)) ?? 
+                (MemberInfo)compiledType.GetField(nameof(IRazorPage.Path));
+            Debug.Assert(pathProperty != null);
```

Comment:
If there is no Path property, it will result in NullReferenceExpeption. Not sure if this will happen at all so added an assertion.